foundry/applications/formApplications/imagePopout.d.ts

```diff
@@ -1,10 +1,96 @@
 /**
- * An Image Popout Application
- * Provides optional support to edit the image path being viewed
- * @param image - The image being viewed
- * @param options - Standard Application rendering options
- * @param onUpdate - An optional callback function which should be triggered if the Image path is edited
+ * An Image Popout Application which features a single image in a lightbox style frame.
+ * This popout can also be used as a form, allowing the user to edit an image which is being used.
+ * Furthermore, this application allows for sharing the display of an image with other connected players.
+ *
+ * @example
+ * ```typescript
+ * // Construct the Application instance
+ * const ip = new ImagePopout("path/to/image.jpg", {
+ *   title: "My Featured Image",
+ *   shareable: true,
+ *   entity: game.actors.getName("My Hero")
+ * });
+ *
+ * // Display the image popout
+ * ip.render(true);
+ *
+ * // Share the image with other connected players
+ * ip.share();
+ * ```
  */
-declare class ImagePopout extends FormApplication {
-  _updateObject(event?: Event, formData?: object): Promise<any>; // TODO: Add correct return type
+declare class ImagePopout extends FormApplication<ImagePopout.Data, string> {
+  constructor(src: string, options?: ImagePopout.Options);
+
+  protected _related: Entity | object | null;
+
+  /** @override */
+  static get defaultOptions(): ImagePopout.Options;
+
+  /** @override */
+  get title(): string;
+
+  /** @override */
+  getData(options?: Application.RenderOptions): ImagePopout.Data | Promise<ImagePopout.Data>;
```

Comment:
This always returns a Promise as far as I can tell, so the return type can be narrowed.
```suggestion
  getData(options?: Application.RenderOptions): Promise<ImagePopout.Data>;
```

Reply:
I think the point of having the return type this wide is to allow people who derive from this class implement it in the way that they want. We did that in a couple of other places. But might not make sense here.